agent/src/main/java/com/microsoft/hydralab/agent/runner/TestRunner.java

```diff
@@ -76,30 +137,8 @@ private void presetForMicrosoftLauncherApp(DeviceInfo deviceInfo, TestTask testT
         ThreadUtils.safeSleep(3000);
     }
 
-    protected void afterTest(DeviceInfo deviceInfo, TestTask testTask, DeviceTestTask deviceTestTask, TestTaskRunCallback testTaskRunCallback, Logger reportLogger) {
-        if (testTaskRunCallback != null) {
-            try {
-                String absoluteReportPath = xmlBuilder.buildTestResultXml(testTask, deviceTestTask);
-                deviceTestTask.setTestXmlReportPath(deviceManager.getTestBaseRelPathInUrl(new File(absoluteReportPath)));
-            } catch (Exception e) {
-                reportLogger.error("Error in buildTestResultXml", e);
-            }
-            try {
-                testTaskRunCallback.onOneDeviceComplete(testTask, deviceInfo, reportLogger, deviceTestTask);
-            } catch (Exception e) {
-                reportLogger.error("Error in onOneDeviceComplete", e);
-            }
-        }
-        deviceManager.testDeviceUnset(deviceInfo, reportLogger);
-        reportLogger.info("Finally: restore state");
-        /* Close/finish resource */
-        if (reportLogger != null) {
-            reportLogger.info("Start Close/finish resource");
-            LogUtils.releaseLogger(reportLogger);
-        }
-    }
 
-    public void reInstallApp(DeviceInfo deviceInfo, TestTask testTask, Logger reportLogger) throws Exception {
+    protected void reInstallApp(DeviceInfo deviceInfo, TestTask testTask, Logger reportLogger) throws Exception {
```

Comment:
Maybe we should replace the implement of this method in parent class and child classes. Only espresso need to install 2 app, the rest like monkey, smart and appium just need 1 apk.
![image](https://user-images.githubusercontent.com/26757995/208354908-11415bb0-5020-49b5-9358-d29d1905922f.png)